arangod/Cluster/DBServerAgencySync.cpp

```diff
@@ -67,110 +136,75 @@ DBServerAgencySyncResult DBServerAgencySync::execute() {
     return result;
   }
   
+  Result tmp;
+  VPackBuilder rb;
   auto clusterInfo = ClusterInfo::instance();
   auto plan = clusterInfo->getPlan();
   auto current = clusterInfo->getCurrent();
-  DatabaseGuard guard(*vocbase);
-  double startTime = TRI_microtime();
-  V8Context* context = V8DealerFeature::DEALER->enterContext(vocbase, true, V8DealerFeature::ANY_CONTEXT_OR_PRIORITY);
+  auto serverId = arangodb::ServerState::instance()->getId();
 
-  if (context == nullptr) {
-    LOG_TOPIC(WARN, arangodb::Logger::HEARTBEAT) << "DBServerAgencySync::execute: no V8 context";
+  DatabaseGuard guard(*vocbase);
```

Comment:
Previous DatabaseGuard object was surrounded by a try/catch block.  Is that needed here too?

Reply:
It can be solved, one needs a try/catch and then put all actions that should be guarded in the try block.